SharedLibs/TileMapHelper/src/model/TileMapModel.ts

```diff
@@ -330,12 +436,58 @@ export class EditableTileMapLayer extends AbstractEditableLayer {
    */
   constructor(tileMap: EditableTileMap, id: integer) {
     super(tileMap, id);
+    this.buildEmptyLayer(
+      this.tileMap.getDimensionX(),
+      this.tileMap.getDimensionY()
+    );
+    this._alpha = 1;
+  }
+
+  buildEmptyLayer(dimensionX: number, dimensionY: number) {
     this._tiles = [];
-    this._tiles.length = this.tileMap.getDimensionY();
+    this._tiles.length = dimensionY;
     for (let index = 0; index < this._tiles.length; index++) {
-      this._tiles[index] = new Int32Array(this.tileMap.getDimensionX());
+      this._tiles[index] = new Int32Array(dimensionX);
     }
-    this._alpha = 1;
+  }
+
+  static from(
+    editableTileMapLayerAsJsObject: any,
+    tileMap: EditableTileMap,
+    isTileIdValid: (tileId: number) => boolean
+  ): EditableTileMapLayer {
+    const layer = new EditableTileMapLayer(
+      tileMap,
+      editableTileMapLayerAsJsObject.id
+    );
+    layer.setAlpha(editableTileMapLayerAsJsObject.alpha);
+    editableTileMapLayerAsJsObject.tiles.forEach((row: Int32Array, y: number) =>
+      row.forEach((tileGID, x) => {
+        const tileId = FlippingHelper.getTileId(tileGID);
+        if (isTileIdValid(tileId)) {
+          layer.setTileGID(x, y, tileGID);
+        }
+      })
+    );
+    return layer;
+  }
+
+  toJSObject(): Object {
```

Comment:
I'm not found of adding serialization helpers directly into the model, but I don't know what it would imply to do it outside.

Reply:
To me, only the object is supposed to know what fields are relevant to serialize or not.

Reply:
In general, I think that a model should have everything needed in its interface to write serialization from the outside. Serialization directly in the objects is something that kind of works until you need to support several file formats.
That said, it's also what Core do.

Reply:
I agree about file formats. In this case it's easy to make it an external serializer. I'll leave it as is.